tools/Update-Library.ps1

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+param(
+	$SdkVersion = $(throw "SdkVersion not provided"),
+	$NewLibraryVersion = $(throw "NewLibraryVersion not provided")
+)
+
+$SrcFolder = "..\src"
+$LibConfigFile = "..\library.properties"
+
+Write-Host "Clonning azure-sdk-for-c repository."
+
+git clone -b $SdkVersion https://github.com/Azure/azure-sdk-for-c sdkrepo
+
+Write-Host "Flattening the azure-sdk-for-c file structure and updating src/."
+
+$Files = gci -Recurse -Include *.h, *.c .\sdkrepo\sdk  | ?{ $_.DirectoryName -INOTMATCH "tests|sample" -AND $_.Name -INOTMATCH "curl|win32|az_posix" }
+
+rm -Force -Exclude "azure_ca.h" $SrcFolder/*
+
+copy -Verbose $Files $SrcFolder
+
+Write-Host "Filtering out files not needed/supported on Arduino."
```

Comment:
nit: Isn't the command on line 18 above doing the filtering?

Line 26, below, is updating include headers within the files based on the flattening.

Reply:
Correct. Fixed with a couple comments in the script.